skimage/measure/_moments_cy.pyx

```diff
@@ -2,17 +2,26 @@
 #cython: nonecheck=False
 #cython: wraparound=False
 import numpy as np
+cimport numpy as cnp
 
+from .._shared.fused_numerics cimport np_floats
+cnp.import_array()
 
-def moments_hu(double[:, :] nu):
-    cdef double[::1] hu = np.zeros((7, ), dtype=np.double)
-    cdef double t0 = nu[3, 0] + nu[1, 2]
-    cdef double t1 = nu[2, 1] + nu[0, 3]
-    cdef double q0 = t0 * t0
-    cdef double q1 = t1 * t1
-    cdef double n4 = 4 * nu[1, 1]
-    cdef double s = nu[2, 0] + nu[0, 2]
-    cdef double d = nu[2, 0] - nu[0, 2]
+
+def moments_hu(np_floats[:, :] nu):
+    if np_floats is cnp.float32_t:
+        dtype = np.float32
+    else:
+        dtype = np.float64
```

Comment:
Can you omit this as in the other PR you opened omits this and just use dtype=nu.dtype below

Reply:
To have access to `nu.dtype` in #4307, `nu` is declared as `cnp.ndarray` in function signature, which is [discouraged]( http://docs.cython.org/en/latest/src/userguide/memoryviews.html#comparison-to-the-old-buffer-support).